src/openvic-simulation/dataloader/NodeTools.cpp

```diff
@@ -234,19 +234,19 @@ node_callback_t NodeTools::expect_date_identifier_or_string(callback_t<Date> cal
 }
 
 node_callback_t NodeTools::expect_years(callback_t<Timespan> callback) {
-	return expect_uint<Timespan::day_t>([callback](Timespan::day_t val) -> bool {
+	return expect_int<Timespan::day_t>([callback](Timespan::day_t val) -> bool {
```

Comment:
Why?

Reply:
The `CAMPAIGN_EVENT_STATE_SCALE` define value is, from what I can understand, a number of days that the event time changes by for each non colonial state, and in vanilla it's `-3` (so each non-colonial state reduces the campaign event time by 3 days). This define has a comment saying it's not used, but nonetheless I decided best to allow days/months/years to be negative so they can represent changes in amount of time if necessary. It might be worth separating these functions into regular versions (zero or positive only) and "vector time" (negative allowed) versions if there are cases where we really need to disallow negative day/month/year quantities.